### PR TITLE
Edit pkgbuild

### PR DESCRIPTION
**Pull request by L0ric0**
added ability to edit the pkgbuild before installing a package.

### CODE REVIEW

File: tests/test_pkgbuilder.py [hunk 1]
```diff
@@ -78,8 +78,8 @@ def test_pbds_logging(self):
         pbds.log.debug('PB unittest/TestPB is running now on this machine.')
 
     def test_utils_print_package_search(self):
-        sample = ('aur/pkgbuilderts 3.2.0-1 (19 votes) '
-                  '\x1b[1;1m\x1b[1;31m[out of date]\x1b[1;0m\n'
+        sample = ('aur/pkgbuilderts 3.2.0-1 (19 Stimmen) '
```
[the remark this answers is not in the document]
Comment (L0ric0) [reply tests/test_pkgbuilder.py]: sorry reverted the commit

File: pkgbuilder/build.py [hunk 2]
```diff
@@ -411,8 +411,28 @@ def fetch_runner(pkgnames, preprocessed=False):
...
+    while done:
+        _confirm = input('\nEdit PKGBUILD of ' + pkgname + '?[Y/n]')
+        if _confirm in ['Y', 'y', 'yes', 'Yes'] or not _confirm:
+            done = False
+            confirm = True
+        elif _confirm in ['N', 'n', 'no', 'No']:
+            done = False
+            confirm = False
+        else:
+            print("Please awnser with 'Y', 'y', 'Yes', 'yes', 'N', 'n', 'No' or 'no'.")
```
[the remark this answers is not in the document]
Comment (L0ric0) [reply pkgbuilder/build.py]: done

File: pkgbuilder/build.py [hunk 3]
```diff
@@ -411,8 +411,28 @@ def fetch_runner(pkgnames, preprocessed=False):
...
+    if confirm:
+        if os.environ['EDITOR']:
+            subprocess.call([os.environ['EDITOR'], './PKGBUILD'])
+        else:
+            subprocess.call(['nano', './PKGBUILD'])
+
+
 def build_runner(pkgname, performdepcheck=True,
-                 pkginstall=True):
+                 pkginstall=True, pkgbuild_edit=True):
```
[the remark this answers is not in the document]
Comment (L0ric0) [reply pkgbuilder/build.py]: done

File: pkgbuilder/__main__.py [hunk 4]
```diff
@@ -95,6 +95,12 @@ def main(source='AUTO', quit=True):
         argopt.add_argument(
             '--nodebug', action='store_true', dest='nodebug',
             help=_('don\'t display debug messages (default)'))
+        argopt.add_argument(
+            '--edit-pkgbuild', action='store_true', dest='edit_pkgbuild',
+            help=_('edit the PKGBUILD (default)'))
```
Comment (Kwpolska) [inline pkgbuilder/__main__.py]: leftover `(default)`?

File: pkgbuilder/build.py [hunk 5]
```diff
@@ -411,8 +411,18 @@ def fetch_runner(pkgnames, preprocessed=False):
         exit(1)
 
 
+def edit_pkgbuild(pkgname):
+    yesno = input('\nEdit PKGBUILD of ' + pkgname + '? [Y/n] ')
```
Comment (Kwpolska) [inline pkgbuilder/build.py]: This should be translatable.
```py
input('\n' + _('Edit PKGBUILD of {0}? [Y/n] ').format(pkgname))
```

**Comment by Kwpolska** [comment]
One more minor thing, `edit_pkgbuild=false` should be added to `pkgbuilder/data/pkgbuilder.ini.skel` in the `[options]` section.

**Comment by Kwpolska** [comment]
Thanks for contributing this! I’ll release a new version this weekend.

**Comment by Kwpolska** [comment]
Released as v4.3.0.

**Pull request merged**